File: src/main/java/at/hrechny/predictionsbot/mapper/UserMapper.java
package at.hrechny.predictionsbot.mapper;

import at.hrechny.predictionsbot.config.MapperConfig;
import at.hrechny.predictionsbot.database.entity.UserEntity;
import at.hrechny.predictionsbot.model.User;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(config = MapperConfig.class)
public interface UserMapper {

  @Mapping(target = "name", source = "username")
  User entityToModel(UserEntity source);

}

File: src/main/kotlin/at/hrechny/predictionsbot/mapper/UserMapper.kt
package at.hrechny.predictionsbot.mapper

import at.hrechny.predictionsbot.database.entity.UserEntity
import at.hrechny.predictionsbot.controller.model.user.UserResponseDto
import jakarta.inject.Singleton

@Singleton
class UserMapper {
    fun entityToModel(source: UserEntity): UserResponseDto =
        UserResponseDto().apply {
            id = source.id
            name = source.username
        }
}
